framework/src/messaging/messaging_common.c

```diff
@@ -103,6 +107,50 @@ int messaging_set_notify_signal(int signo, _sa_sigaction_t func)
 	}
 	return OK;
 }
+/****************************************************************************
+ * Name : messaging_parse_packet
+ *
+ * Description:
+ *  Parse the received packet.
+ ****************************************************************************/
+int messaging_parse_packet(char *packet, char *buf, int buflen, pid_t *sender_pid, int *msg_type)
+{
+	int my_version;
+	int msg_version;
+	int parsing_version;
+	int ret = OK;
+	int offset;
+
+	my_version = messaging_get_version();
+
+	/* Received packet(version 1) is like below.
```

Comment:
If the version 1 should follow below type, you had better define structure.
```
struct packet_header_s {
     uint32_t version;
     uint32_t offset;
     uint32_t sender_pid;
     uint32_t msg_type;
}
...
struct message_packet {
    struct packet_header_s header;
    data....
}
....
#define offsetof(type, member) ((size_t)&(((type *)0)->member))
...
case 1 :
         memcpy(sender_pid, offsetof(struct packet_header_s, sender_pid, sizeof(pid_t));
...
```

Reply:
Thanks. It is more clear.
I found that we already have offsetof in <stddef.h>.
I modified.